lib/irb/source_finder.rb

```diff
@@ -69,10 +69,19 @@ def initialize(irb_context)
 
     def find_source(signature, super_level = 0)
       case signature
-      when /\A(::)?[A-Z]\w*(::[A-Z]\w*)*\z/ # Const::Name
+      when /\A(::)?[A-Z]\w*(::[A-Z]\w*)*\z/ # ConstName, ::ConstName, ConstPath::ConstName
         eval_receiver_or_owner(signature) # trigger autoload
-        base = @irb_context.workspace.binding.receiver.yield_self { |r| r.is_a?(Module) ? r : Object }
-        file, line = base.const_source_location(signature)
+        *parts, name = signature.split('::', -1)
+        base = (
```

Comment:
nit: I think having these parentheses introduces a new style that didn't exist in this repo before 🤔 ?

Reply:
Updated to
```ruby
a =
  if cond
  else
  end
```
style